src/wp-includes/compat.php

```diff
@@ -375,6 +375,40 @@ function is_iterable( $var ) {
 	}
 }
 
+if ( ! function_exists( 'str_starts_with' ) ) {
+	/**
+	 * Polyfill for array_key_last() function added in PHP 7.3.
+	 *
+	 * Performs a case-sensitive check indicating if needle is contained in haystack.
+	 *
+	 * @param $haystack The string to search in.
+	 * @param $needle The substring to search for in the haystack to is it startswith it.
+	 *
+	 * @return Boolean true if needle is the haystack and starts with it.
+	 * @since 5.9.0
+	 *
+	 */
```

Comment:
This DocBlock is not correct.  Looks like copy-paste errors from the PR for the `array_key_last()` polyfill PR :-)

Also, some (but not all) of the DocBlocks for pollyfills in wp-includes/compat.php add the `@ignore` tag.  I assume that is so that `phpdoc-parser` skips them and they don't show up in the Code Reference.  For instance, `hash_hmac()` has `@ignore` (and doesn't appear in the Code Reference) but `hash_equals()` does **not** have `@ignore` and [does appear](https://developer.wordpress.org/reference/functions/hash_equals/) in the Code Reference.  Should `str_starts_with()` have `@ignore`??  I'm not sure.

Next, isn't the convention for DocBlocks to have the `@since` tag above the first `@param` tag (with a blank line after it) and for the `@return` tag to **not** have a blank line separating it from the last `@param` tag?

Last, I think it would be a good idea if the descriptions in the `@param` and `@return` tags matched what was in the PHP manual (especially if the `@ignore` tag is not going to be used).  Just a thought.